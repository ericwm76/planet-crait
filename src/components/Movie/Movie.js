import React from 'react';
import './Movie.scss';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const Movie = ({id, title, year, url, img, selectMovie}) => {
  return (
    <section className='movie' onClick={() => selectMovie(url)}>
      <img src={img} alt={title} />
      <h3>Episode {id}:</h3>
      <h2>{title}</h2> 
      <h3>{year}</h3>
      <Link to={`/movies/${id}`}>View Characters</Link>
    </section>
  )  
}

export default Movie;

Movie.propTypes = {
  id: PropTypes.number,
  title: PropTypes.string,
  year: PropTypes.string,
  url: PropTypes.string,
  img: PropTypes.string,
  selectMovie: PropTypes.func
}