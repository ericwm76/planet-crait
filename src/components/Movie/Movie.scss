$crait-color : #c0392b;
 
 .movie {
  margin: 10px;
  padding: 10px;
  border: 2px solid $crait-color;
  border-radius: 10px;
//   height: 50%;
 }

 img {
  height: 17em;
 }