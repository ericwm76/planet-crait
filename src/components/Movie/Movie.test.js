import React from 'react';
import { shallow } from 'enzyme';
import Movie from './Movie';

describe('Movie', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Movie />)
  })

  it('should match snapshot with all data passing in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })
})