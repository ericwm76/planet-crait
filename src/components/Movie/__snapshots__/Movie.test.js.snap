// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie should match snapshot with all data passing in correctly 1`] = `
<section
  className="movie"
  onClick={[Function]}
>
  <img />
  <h3>
    Episode 
    :
  </h3>
  <h2 />
  <h3 />
  <Link
    to="/movies/undefined"
  >
    View Characters
  </Link>
</section>
`;
